dockerHashing/src/main/scala/cromwell/docker/registryv2/flows/HttpFlowWithRetry.scala

```diff
@@ -143,12 +146,24 @@ case class HttpFlowWithRetry[T](
     * Create a re-submittable request from a failed retryable
     * @return a future that will complete after the appropriate backoff time
     */
-  private def toRetryableRequest(value: (Any, ContextWithRequest[T])) = value match {
-    case (_, contextWithRequest) => 
+  private def toRetryableRequest(value: (HttpResponse, ContextWithRequest[T])) = value match {
+    case (response, contextWithRequest) => 
       val nextRetryIn = contextWithRequest.retryIn
       val nextRequest = (contextWithRequest.request, contextWithRequest.withNextAttempt)
-      akka.pattern.after(nextRetryIn, scheduler) { 
-        Future.successful(nextRequest)
+      // This response will never be consumed by anyone, so discard its content here to avoid pool freeze
+      // http://doc.akka.io/docs/akka-http/10.0.5/scala/http/client-side/request-level.html#using-the-future-based-api-in-actors
+      // https://github.com/akka/akka/issues/19538
+      // https://github.com/akka/akka-http/issues/183
+      // https://github.com/akka/akka-http/issues/117
+      response.discardEntityBytes().future() flatMap { _ =>
```

Comment:
Hmm I don't think so because `akka.pattern.after(nextRetryIn, scheduler)` both takes and returns a `Future` so it needs to be a `flatMap`